src/pages/ProviderEServiceCreatePage/components/EServiceCreateStepPurpose/EServiceCreateFromTemplateStepPurpose.tsx

```diff
@@ -0,0 +1,103 @@
+import { SectionContainer, SectionContainerSkeleton } from '@/components/layout/containers'
+import { Button, Stack } from '@mui/material'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useEServiceCreateContext } from '../EServiceCreateContext'
+import { RHFAutocompleteSingle } from '@/components/shared/react-hook-form-inputs'
+import { FormProvider, useForm } from 'react-hook-form'
+import { StepActions } from '@/components/shared/StepActions'
+import ArrowBackIcon from '@mui/icons-material/ArrowBack'
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward'
+import { EServiceRiskAnalysisInfoSummary } from '@/components/shared/RiskAnalysisInfoSummary'
+
+type RiskAnalysisValue = { label: string; value: string }
+
+export const EServiceCreateFromTemplateStepPurpose: React.FC = () => {
+  const { t } = useTranslation('eservice', { keyPrefix: 'create' })
+  const { t: tCommon } = useTranslation('common')
+
+  const { descriptor, forward, back } = useEServiceCreateContext()
+
+  const methods = useForm()
+
+  const riskAnalysisList = descriptor?.eservice.riskAnalysis ?? []
+
+  const autocompleteOptions = (riskAnalysisList ?? []).map((riskAnalysis) => ({
+    label: riskAnalysis.name,
+    value: riskAnalysis.id,
+  }))
+
+  const [selectedRiskAnalysis, setSelectedRiskAnalysis] = useState<RiskAnalysisValue>({
+    label: '',
+    value: '',
+  })
+  const [showRiskAnalysis, setShowRiskAnalisys] = useState(false)
+
+  const resetStates = () => {
+    setShowRiskAnalisys(false)
+    setSelectedRiskAnalysis({ label: '', value: '' })
+  }
+
+  const setStates = (value: RiskAnalysisValue) => {
+    setShowRiskAnalisys(false)
+    setSelectedRiskAnalysis(value)
+  }
+
+  return (
+    <FormProvider {...methods}>
```

Comment:
I would wrap with `FormProvider` only the input and the button, the rest is not part of the form